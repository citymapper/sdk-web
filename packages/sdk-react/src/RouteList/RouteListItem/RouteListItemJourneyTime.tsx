import { DateTime } from 'luxon'
import React from 'react'
import IconArrowRight from '../../Icons/IconArrowRight'
import styled from '../../utils/cmStyled'

interface RouteListItemJourneyTimeProps {
  /**
   * Journey start time in ISO format
   */
  start: string
  /**
   * Journey end time in ISO format
   */
  end: string
  /**
   * React Component to use as the separator between two times
   */
  IconSeparator?: React.ReactElement
}

const Wrapper = styled('div', { className: 'cm-route-times' })`
  display: flex;
  align-items: center;
  font-size: 0.8125em;
  font-weight: 600;
  line-height: 16px;
  margin-top: ${({ theme }) => theme.spacing(1.75)};

  svg {
    width: 13px;
    height: 13px;
    color: ${({ theme }) => theme.palette.tone.grey70.color};
    margin: 0 ${({ theme }) => theme.spacing(0.5)};
    opacity: 0.4;
  }

  &.cm-route-times {
    ${({ theme }) => theme.CmRouteTimes?.root}
  }
`

const RouteListItemJourneyTime: React.FC<RouteListItemJourneyTimeProps> = ({
  start,
  end,
  IconSeparator,
}) => {
  return (
    <Wrapper>
      {DateTime.fromISO(start).toLocaleString(DateTime.TIME_24_SIMPLE)}
      {IconSeparator ? IconSeparator : <IconArrowRight />}
      {DateTime.fromISO(end).toLocaleString(DateTime.TIME_24_SIMPLE)}
    </Wrapper>
  )
}

export default RouteListItemJourneyTime
