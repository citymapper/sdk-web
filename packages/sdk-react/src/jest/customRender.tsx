import { render, RenderOptions } from '@testing-library/react'
import React, { JSXElementConstructor, ReactElement } from 'react'
import CMProvider from '../CMProvider'

const TestWrapper: React.FC<{ children: React.ReactElement }> = ({
  children,
}) => (
  <CMProvider
    config={{
      apiKey: '',
      citymapperFont: false,
    }}
  >
    {children}
  </CMProvider>
)

const customRender = (
  ui: ReactElement<any, string | JSXElementConstructor<any>>,
  options?: RenderOptions<
    typeof import('@testing-library/dom/types/queries'),
    HTMLElement,
    HTMLElement
  >
) => render(ui, { wrapper: TestWrapper, ...options })

export default customRender
