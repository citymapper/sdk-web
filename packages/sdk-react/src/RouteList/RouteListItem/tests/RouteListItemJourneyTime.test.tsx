import React from 'react'
import customRender from '../../../jest/customRender'
import RouteListItemJourneyTime from '../RouteListItemJourneyTime'

describe('RouteListItemJourneyTime', () => {
  it('should match snapshot', () => {
    const { container } = customRender(
      <RouteListItemJourneyTime
        start="2022-07-12T14:56:19+01:00"
        end="2022-07-12T14:46:19+01:00"
      />
    )
    expect(container).toMatchSnapshot()
  })
})
