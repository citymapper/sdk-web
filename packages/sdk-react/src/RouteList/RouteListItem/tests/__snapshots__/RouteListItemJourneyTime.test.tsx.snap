// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteListItemJourneyTime should match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 0.8125em;
  font-weight: 600;
  line-height: 16px;
  margin-top: 14px;
}

.emotion-0 svg {
  width: 13px;
  height: 13px;
  color: #424750;
  margin: 0 4px;
  opacity: 0.4;
}

.emotion-1 {
  width: 24px;
  height: 24px;
  width: 24;
  height: 24;
}

<div>
  <div
    class="cm-route-times emotion-0"
  >
    14:56
    <svg
      class="emotion-1"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        clip-rule="evenodd"
        d="M12.7626 3.76256C13.446 3.07915 14.554 3.07915 15.2374 3.76256L22.2374 10.7626C22.9209 11.446 22.9209 12.554 22.2374 13.2374L15.2374 20.2374C14.554 20.9209 13.446 20.9209 12.7626 20.2374C12.0791 19.554 12.0791 18.446 12.7626 17.7626L16.7751 13.75H3C2.0335 13.75 1.25 12.9665 1.25 12C1.25 11.0335 2.0335 10.25 3 10.25H16.7751L12.7626 6.23744C12.0791 5.55402 12.0791 4.44598 12.7626 3.76256Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
    14:46
  </div>
</div>
`;
